### ISSUE TITLE
torch.nn.utils.rnn.pad_sequence replaces entire output tensor with padding_value on TPU

### ISSUE THREAD

**Issue by hjajoo-g** [google/torchax#82]
Using torch.nn.utils.rnn.pad_sequence on a TPU via Torchax results in an output tensor that is entirely filled with the padding_value. Instead of just padding the ends of shorter sequences, the operation fails to copy any of the input data into the output buffer, leading to complete data loss.

**Reproduction Script:** This script demonstrates that the output is entirely replaced by whatever padding_value is provided (confirmed with both 0.0 and -1.0).

```
import torch
import torchax

torchax.enable_globally()

def reproduce(padding=-1.0):
    print(f"Testing pad_sequence on TPU with padding_value={padding}")
    
    # Create test sequences with non-zero values
    t1 = torch.tensor([1.0, 2.0, 3.0])
    t2 = torch.tensor([4.0, 5.0])
    sequences = [t1, t2]
    
    # Move to jax device
    device = 'jax'
    tensors_jax = [t.to(device) for t in sequences]
    
    # Apply pad_sequence
    padded = torch.nn.utils.rnn.pad_sequence(tensors_jax, batch_first=True, padding_value=padding)
    
    print("\nOutput Tensor:")
    print(padded)
    
    if torch.all(padded == padding):
        print(f"\n[RESULT] BUG REPRODUCED: The entire tensor was replaced by {padding}!")

if __name__ == "__main__":
    reproduce(padding=-1.0)
    reproduce(padding=0.0)
```

**Expected Result:** The output tensor should contain the original sequence data with padding at the end:
```
tensor([[ 1.,  2.,  3.],
        [ 4.,  5., -1.]], device='jax:0')
```

**Actual Result:** The entire tensor is filled with the padding value, and all input data is lost:
```
tensor([[-1., -1., -1.],
        [-1., -1., -1.]], device='jax:0')
```
